File: backend/app/schemas/exam.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ExamBase(BaseModel):
    title: str
    description: Optional[str] = None
    start_time: datetime
    end_time: datetime
    duration: int
    total_score: int = 100
    pass_score: int = 60
    random_order: bool = True
    max_switch: int = 3

class ExamCreate(ExamBase):
    course_id: int

class ExamUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    duration: Optional[int] = None
    total_score: Optional[int] = None
    pass_score: Optional[int] = None
    random_order: Optional[bool] = None
    max_switch: Optional[int] = None
    status: Optional[str] = None

class ExamResponse(ExamBase):
    id: int
    course_id: int
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/exam.ts
import type { Dayjs } from 'dayjs';

export type ExamStatus = 'draft' | 'published' | 'ongoing' | 'finished';

export interface Exam {
  id: number;
  course_id: number;
  title: string;
  description?: string | null;
  start_time: string;
  end_time: string;
  duration: number;
  total_score: number;
  pass_score: number;
  random_order: boolean;
  max_switch: number;
  status: ExamStatus;
  created_at: string;
}

export interface ExamInput {
  course_id?: number;
  title: string;
  description?: string;
  duration: number;
  total_score: number;
  pass_score: number;
  start_time: string;
  end_time: string;
  random_order: boolean;
  max_switch: number;
}

// 考试编辑页表单值（DatePicker 值，提交前 format 成字符串）
export interface ExamFormValues {
  course_id?: number;
  title: string;
  description?: string;
  duration: number;
  total_score: number;
  pass_score: number;
  max_switch: number;
  start_time: Dayjs;
  end_time: Dayjs;
  random_order: boolean;
}

export interface ExamQuery {
  page?: number;
  page_size?: number;
  status?: string;
  course_id?: number;
}
